httpbp/errors.go

```diff
@@ -0,0 +1,221 @@
+package httpbp
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+)
+
+// HTTPError is an error that and can be returned by an  HTTPHandler to return a
+// customized error response.
+type HTTPError interface {
+	error
+
+	// Response returns the custom Response for the error to be written by
+	// the ContentWriter.
+	Response() Response
+
+	// ContentWriter returns the ContentWriter object to use to write the error
+	// response.
+	ContentWriter() ContentWriter
+
+	// Unwrap implements helper interface for errors.Unwrap.  Should return the
+	// internal error that triggered the HTTPError to be returned to the caller.
+	Unwrap() error
+}
+
+// NewHTTPError returns a new HTTPError object initialized with the given
+// values.
+//
+// NewHTTPError is provided for testing purposes and should not be used directly,
+// you should use one of the JSON/HTML/Raw error functions to return standard
+// HTTPErrors
+func NewHTTPError(code int, body interface{}, cause error, cw ContentWriter) HTTPError {
```

Comment:
🔕 Is it ever used outside of this package? Or do you expect service code might need to use it in their own unit tests?

If not we can probably unexport it.

Reply:
We _probably_ don't need to export it anymore.